classes/timeslot_manager.php

```diff
@@ -427,12 +432,50 @@ public static function send_signup_confirmation_message(int $observationid, int
         return message_send($eventdata);
     }
 
+    /**
+     * Sends a reminder message about a timeslot to a specific user.
+     * @param int $observationid id of the observation instance.
+     * @param int $slotid id of the timeslot the user signed up to.
+     * @param int $userid id of the user to send the message to.
+     */
+    public static function send_reminder_message(int $observationid, int $slotid, int $userid) {
+        global $DB;
+        $user = $DB->get_record('user', ['id' => $userid]);
+
+        list($observation, $course, $cm) =
+            \mod_observation\observation_manager::get_observation_course_cm_from_obid($observationid);
+
+        $contexturl =
+            (new \moodle_url('/mod/observation/timeslotjoining.php', ['id' => $observation->id]))->out(false);
+
+        $eventdata = new \core\message\message();
+
+        $eventdata->courseid          = $course->id;
+        $eventdata->component         = 'mod_observation';
+        $eventdata->name              = 'signupreminder';
+        $eventdata->notification      = 1;
+
+        $eventdata->userfrom          = \core_user::get_noreply_user();
```

Comment:
like the cancellation message, should this be from the observer instead? that would be the likely contact for any problems